### ISSUE TITLE
Changeable Port

### ISSUE THREAD

**Issue by JxxIT** [jomjol/AI-on-the-edge-device#2707]
### The Feature

First off, thanks for this amazing tool!

This feature adds a expert setting to change the port off the web interface. Working on a PR.

**Comment by caco3** [comment]
I think this is really expert feature and would cause a lot of trouble to normal users.
If you really need it, you easily con modify the firmware and build it yourself.

**Comment by JxxIT** [comment]
@caco3 
Honestly, I completely forgot about this. 😄 

In my case, I wanted to port forward both of my devices. My router doesn't have an option to change the change ports, so I had to change it manually.

> I think this is really expert feature and would cause a lot of trouble to normal users.

Can you please explain why it is a expert feature and would cause a lot of trouble? I really missed this feature and I think a lot of users would like to see this.

**Comment by SybexX** [comment]
@JxxIT It wouldn't have helped you either, since the ESP couldn't connect and you couldn't access the website^^

you can take a look there: https://github.com/jomjol/AI-on-the-edge-device/discussions/2147#discussioncomment-10423313

**Comment by JxxIT** [comment]
@SybexX Do you mean by changing it directly in the source code? I did that, built, and it worked.

**Comment by SybexX** [comment]
I meant if you could see the IP on the web interface.
Directly in the source code is not possible because everyone would have to compile their own firmware and an OTA update would overwrite this.

**Comment by caco3** [comment]
> Can you please explain why it is a expert feature and would cause a lot of trouble? I really missed this feature and I think a lot of users would like to see this.

If it would be a user configurable option, a user, not understanding what he is doing, could change the port number and then lock himself out of the UI as he might not know how to add the port to the URL or forget the exotic number.

If you would want it, you could add a PR but it would have to be set in the wlan.ini file so users can not change it by accident.

**Comment by SybexX** [comment]
I don't know much about port forwarding, but isn't this done by the router/server?

**Comment by caco3** [comment]
Normally yes. Eg. the router can route the external port 12345 to the internal port 80 of a specific IP.
But some routers do not support port forwarding or do not allow to use different ports externally/internally.
But here I must say that it can not be an opensource's responsibility to work around a shitty router.
Get a better router or add a Raspberry Pi as reverse proxy which can do this.

**Comment by SybexX** [comment]
As far as I know, it's switched off on the Fritzbox because it's a security risk or something.

**Comment by caco3** [comment]
Yes, obviously it rips a hole into your network. You should only use it when understanding the impact.
The nowadays better approach is to VPN into your network and access it that way from outside.

**Comment by SybexX** [comment]
I use two Rpi's with iobroker + Adguard that handle everything and don't allow remote access.
